File: Go/Algorithms/Searching/linear_search.go
package main

import "fmt"

func linear_search(numbers []int, input_length int, target int) int {
  for idx := 0; idx < input_length; idx++ {
    if numbers[idx] == target {
      return idx
    }
  }
  return -1
}

func main() {
  // Get inputs
  input_length := 0
  fmt.Println("Enter the number of inputs:")
  fmt.Scanln(&input_length)
  fmt.Println("Enter the inputs:")
  numbers := make([]int, input_length)
  for i := 0; i < input_length; i++ {
      fmt.Scanln(&numbers[i])
  }

  // Get search number
  target := 0
  fmt.Println("Search number:")
  fmt.Scanln(&target)

  // linear search
  result := linear_search(numbers, input_length, target)

  if result == -1 {
    fmt.Println("Target not found")
  } else {
    fmt.Println("Target", target, "is at position", result)
  }
}

File: C/Algorithms/Searching/Linear_Search.c
#include<stdio.h>
#include<stdlib.h>

int linearSearch(int a[], int n, int length);

void main()
{
    int length, i, find;
    printf("Linear Search\n");
    printf("How long of the array do you want it to be? >> ");
    scanf("%d", &length);
    
    int array[length];

    printf("Now you can enter the integers!\n");
    for ( i = 0; i < length; i++)
    {
    	printf("%d : ", i);
        scanf("%d", &array[i]);
    }
    
    printf("Now which number do you want to find? >> ");
    scanf("%d", &find);
    printf("%d is at %d", find, linearSearch(array, find, length));
}

int linearSearch(int a[], int n, int length)
{
    int i;
    for(i = 0; i<length; i++){
    	if(a[i] == n) return i;
	}
}
